src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,178 +3,108 @@
+    private static final Logger log = LoggerFactory.getLogger(IssuesAPICaller.class);
     private static final ObjectMapper objectMapper = new ObjectMapper();
+    private final OkHttpClient client;
     private final String authentication;
-    private String jsessionId = null;
-    private String upstream = null;
+
     private String cookies = "";
     private String pinnedNode = null;
 
     public ApiCaller(final JiraActionsIndexBuilderConfig config) {
         this.config = config;
         this.authentication = getBasicAuth();
+        this.client = new OkHttpClient.Builder().build();
     }
 
     public JsonNode getJsonNode(final String url) throws IOException {
-        HttpsURLConnection urlConnection = null;
-        Map<String, List<String>> headers = null;
-        BufferedReader br = null;
-        String apiResults = null;
-        try {
-            urlConnection = getURLConnection(url);
-            headers = urlConnection.getRequestProperties();
-            final InputStream in = urlConnection.getInputStream();
-            br = new BufferedReader(new InputStreamReader(in));
-            apiResults = br.readLine();
-
-            final String anodeId = urlConnection.getHeaderField("X-ANODEID");
-
-            if(!Objects.equals(pinnedNode, anodeId)) {
-                if(pinnedNode != null) {
-                    log.warn("Expected X-ANODEID={} but found {}", pinnedNode, anodeId);
-                }
-                final Map<String, List<String>> responseHeaders = urlConnection.getHeaderFields();
-                final List<String> cookies = responseHeaders.get("Set-Cookie");
-                if(cookies != null) {
-                    for (final String cookie : cookies) {
-                        if (cookie.startsWith("JSESSIONID=")) {
-                            final int start = "JSESSIONID=".length();
-                            final int end = cookie.contains(";") ? cookie.indexOf(";") : cookie.length();
-                            jsessionId = cookie.substring(start, end);
-                        } else if (cookie.startsWith("upstream")) {
-                            final int start = "upstream=".length();
-                            final int end = cookie.contains(";") ? cookie.indexOf(";") : cookie.length();
-                            upstream = cookie.substring(start, end);
-                        }
-                    }
-                    if (jsessionId != null || upstream != null) {
-                        setCookies();
-                        pinnedNode = anodeId;
-                        log.info("Set JSESSION={};upstream={}. Pinning to X-ANODEID={}",
-                                jsessionId, upstream, anodeId);
-                    }
-                }
+        final Request request = new Request.Builder()
+                .header("Authorization", this.authentication)
+                .header("Cache-Control", "no-store")
+                .header("Cookie", cookies)
```

Comment:
Manually handling the cookies seems way more cumbersome than just implementing a CookieJar